### ISSUE TITLE
Sticker database is reset on database update

### ISSUE THREAD

**Issue by tae-soo-kim** [MusicPlayerDaemon/MPD#1897]
### Describe the bug

1. Start MPD
2. Write to sticker database
3. Stop MPD
4. Change `music_directory` in `mpd.conf`
5. Start MPD
6. Database update
7. Read from sticker database

## Expected Behavior

Old sticker database data shall be kept available.

This behavior is expected for two use cases:

1. Old and new music directories have files in common so sticker data can be shared.
2. Switch between two music directories occasionally, without losing any sticker data.

## Actual Behavior

Old sticker data is lost. Sticker database is empty after database update.

One may consider old sticker data not applicable to new music directory. But this is not always true. Accidentally losing all sticker data on a database update is a bug in my opinion.

## Version

Music Player Daemon 0.23.14 (0.23.14)

**Comment by MaxKellermann** [comment]
If you switch music directories, you must also switch sticker databases.
If a song file disappears, MPD will remove any stickers attached to it. If MPD wouldn't do that, your sticker database would grow forever, and then you'd complain that it still contain stickers for songs that you deleted 10 years ago, wouldn't you?
What is your exact suggestion, and how would you respond to users complaining about the behavior you suggest?

**Comment by tae-soo-kim** [comment]
> What is your exact suggestion, and how would you respond to users complaining about the behavior you suggest?

Thanks for clarification. I didn't find anything about this issue in the [doc](https://mpd.readthedocs.io/en/latest/user.html#the-sticker-database). What happened to me was the whole sticker database suddenly gone without a notice and I lost all the data inside by switching the music directory. Next time when I switch the music directory again, I would switch the stickers too. But I think it's good for users to have a `keep_sticker` config option so that they can decide by themselves when to remove stickers. This option is currently *false* and they don't complain if they keep it *false* and everything remains the same. For some use cases, for example when the music directory has manageable total size but needs to be frequently switched, setting `keep_sticker` to *true* could be helpful.

**Comment by MaxKellermann** [comment]
Yes, that sounds like a good feature! Let's change this to a feature request then.

**Comment by v-Nyo** [comment]
+1 for this. I think many people on rmpc use the sicker db. 

I accidentally nuked most of data when I did not mount all drives :(
